packages/react-components/src/atoms/Markdown.tsx

```diff
@@ -5,13 +5,27 @@ import remark2rehype from 'remark-rehype';
 import markdown from 'remark-parse';
 import rehypeReact from 'rehype-react';
 import rehypeSanitize from 'rehype-sanitize';
+import rehypeSlug from 'rehype-slug';
+import rehypeToc from 'rehype-toc';
 import { parseComponents, parseTagNames } from '../utils/parsing';
 
-const Markdown = ({ value }: { value: string }) => {
-  const processor = unified()
-    .use(markdown)
-    .use(remark2rehype)
+interface MarkdownProps {
+  value: string;
+  toc?: boolean;
+}
+
+const Markdown = ({ value, toc = false }: MarkdownProps) => {
+  let processor = unified().use(markdown).use(remark2rehype);
+
+  if (toc) {
+    processor = processor
```

Comment:
Do we need to reassign the processor?
```suggestion
    processor
      .use(rehypeSlug)
      .use(rehypeToc, { headings: ['h1', 'h2'] });
```